modules/mgidBidAdapter.js

```diff
@@ -0,0 +1,232 @@
+   * @return ServerRequest Info describing the request to the server.
+   */
+  buildRequests: (validBidRequests, bidderRequest) => {
+    utils.logInfo(`MGID DEBUG: buildRequests`);
+    if (validBidRequests.length == 0) {
+      return null;
+    }
+    const referer = utils.deepAccess(bidderRequest, 'refererInfo.referer');
+    const hostname = urlUtils.parse(referer).hostname;
+    let domain = extractDomainFromHost(hostname) || hostname;
+    const accountId = setOnAny(validBidRequests, 'params.accountId');
+    const muid = getLocalStorageSafely('mgMuidn');
+    let url = (setOnAny(validBidRequests, 'params.bidUrl') || ENDPOINT_URL) + accountId;
+    if (muid != null && typeof (muid) == 'string' && muid.length > 0) {
+      url += '?muid=' + muid;
+    }
+    const page = utils.deepAccess(bidderRequest, 'refererInfo.canonicalUrl') || referer;
+    const secure = window.location.protocol === 'https:' ? 1 : 0;
+    const imp = validBidRequests.map((bid, id) => {
+      const placeId = utils.deepAccess(bid, 'params.placementId');
+      const sizes = utils.deepAccess(bid, 'mediaTypes.banner.sizes');
+      let format = [];
+      if (sizes.length > 1) {
+        for (let f = 0; f < sizes.length; f++) {
+          if (sizes[f].length == 2) {
+            format.push({w: sizes[f][0], h: sizes[f][1]});
+          }
+        }
+      }
+      return {
+        id: bid.bidId,
+        tagid: placeId,
+        banner: {
+          w: sizes && sizes[0][0],
+          h: sizes && sizes[0][1],
+          format,
+        },
+        secure,
+      };
+    });
+
+    const request = {
+      id: utils.deepAccess(bidderRequest, 'bidderRequestId'),
+      site: { domain, page },
+      cur: ['USD'],
+      device: {
+        ua: navigator.userAgent,
+        js: 1,
+        dnt: (navigator.doNotTrack == 'yes' || navigator.doNotTrack == '1' || navigator.msDoNotTrack == '1') ? 1 : 0,
+        h: screen.height,
+        w: screen.width,
+        language: getLanguage()
+      },
+      ext: {ver: VERSION},
+      imp
+    };
+    utils.logInfo(`MGID DEBUG: buildRequests\n${request}`);
+    return {
+      method: 'POST',
+      url: url,
+      data: JSON.stringify(request),
+    };
+  },
+  /**
+   * Unpack the response from the server into a list of bids.
+   *
+   * @param {ServerResponse} serverResponse A successful response from the server.
+   * @return {Bid[]} An array of bids which were nested inside the server.
+   */
+  interpretResponse: (serverResponse, bidRequests) => {
+    if (serverResponse == null || serverResponse.body == null || serverResponse.body == '' || !serverResponse.body.seatbid || !serverResponse.body.seatbid[0]) {
+      return [];
+    }
+    utils.logInfo(`MGID DEBUG: interpretResponse`);
+    const returnedBids = [];
+    const muidn = utils.deepAccess(serverResponse.body, 'ext.muidn')
+    if (muidn != null && typeof (muidn) == 'string' && muidn.length > 0) {
+      setLocalStorageSafely('mgMuidn', muidn)
```

Comment:
Just making sure, but is this adapter only going to be used in non-GDPR environments? Otherwise, to save user data locally, the consentManagement module needs to be checked for consentString and the vendors list for consent to save data locally. If this isn't going to be used in a GDPR environment, this should be ok.

Reply:
added support of gdpr. we'll not return ext.muidn if tracking is not allowed

Reply:
@idettman added test params to docs